src/renderer/views/Watch/Watch.js

```diff
@@ -876,16 +876,37 @@ export default Vue.extend({
           }
         }, nextVideoInterval * 1000)
 
-        this.showToast({
-          message: this.$tc('Playing Next Video Interval', nextVideoInterval, { nextVideoInterval: nextVideoInterval }),
-          time: (nextVideoInterval * 1000) + 500,
-          action: () => {
-            clearTimeout(this.playNextTimeout)
-            this.showToast({
-              message: this.$t('Canceled next video autoplay')
-            })
+        let countDownTimeLeftInMs = (nextVideoInterval * 1000)
+        let countDownTimeLeftInSecond = nextVideoInterval
+        const showCountDownMessage = () => {
+          // Will not display "Playing next video in no time" as it's too late to cancel
+          // Also there is a separate message when playing next video
+          if (countDownTimeLeftInMs <= 0) {
+            clearInterval(countDownIntervalId)
+            return
           }
-        })
+
+          this.showToast({
+            message: this.$tc('Playing Next Video Interval', countDownTimeLeftInSecond, { nextVideoInterval: countDownTimeLeftInSecond }),
+            // To avoid message flashing
+            // `time` is manually tested to be 700
+            time: 700,
+            action: () => {
+              clearTimeout(this.playNextTimeout)
+              clearInterval(countDownIntervalId)
+              this.showToast({
+                message: this.$t('Canceled next video autoplay')
+              })
+            }
+          })
+
+          countDownTimeLeftInMs = countDownTimeLeftInMs - 1000
```

Comment:
This variable is completely unnecessary.
You can just use `countDownTimeLeftInSecond` in the conditional logic above.